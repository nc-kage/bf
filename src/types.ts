export type OrderType = [number, [number, number, number]];
export type OrdersType = [number, [number, number, number][]];

export type JsonType<P = null | string | number | boolean> = {
  [key: string]: P | JsonType<P> | JsonType<P>[];
};

export enum OrderSymbol {
  BTCUSD = 'tBTCUSD',
  ETHUSD = 'tETHUSD',
}

export enum OrderProposition {
  BID = 1,
  ASK = -1,
}

export enum MessageEvent {
  SUBSCRIBE = 'subscribe',
  SUBSCRIBED = 'subscribed',
  UNSUBSCRIBE = 'unsubscribe',
  UNSUBSCRIBED = 'unsubscribed',
}

export enum Channel {
  BOOK = 'book',
}
