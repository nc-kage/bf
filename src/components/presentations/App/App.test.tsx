import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import Comp from '.';
import { OrderSymbol } from '@/types';

jest.mock('@/components/containers/Orders', () => (): null => null);

describe('App', () => {
  it('snapshot', () => {
    const { asFragment } = render(
      <Comp
        symbols={[OrderSymbol.BTCUSD, OrderSymbol.ETHUSD]}
        selectedSymbol={OrderSymbol.BTCUSD}
        onSelectSymbol={jest.fn()}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
