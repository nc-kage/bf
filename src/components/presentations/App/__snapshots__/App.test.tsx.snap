// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<DocumentFragment>
  <select
    data-testid="select-symbol"
  >
    <option
      data-testid="select-symbol-tBTCUSD"
      value="tBTCUSD"
    >
      BTC/USD
    </option>
    <option
      data-testid="select-symbol-tETHUSD"
      value="tETHUSD"
    >
      ETH/USD
    </option>
  </select>
  <div
    class="App"
  >
    <div
      class="App__header"
    >
      <div>
        Order book 
        <span
          class="App__header__symbol"
        >
          BTC/USD
        </span>
      </div>
    </div>
    <div
      class="App__charts"
    />
  </div>
</DocumentFragment>
`;
